dev_docs/key_concepts/encrypted_saved_objects.mdx

```diff
@@ -0,0 +1,175 @@
+
+When making the decision of which attributes to include in AAD, it is best to be conservative and only include attributes that the owning team is 100% confident
+should be included.
+
+## Caveats
+
+### Partial Updates
+
+Partial updates on ESOs are only possible if the changes are limited to unencrypted and non-AAD attributes. Any changes to an ESO's encrypted values or AAD-
+included values requires re-encryption, which means the entire object must be provided when updating to avoid corrupting the object. If an ESO is corrupted by
+a partial update, it will be effectively undecryptable.
+
+### Migrations, Backward compatibility, and Serverless
+
+With time you may need to change your ESO types. Model Versions allow developers to make versioned changes to Saved Object types, but an ESO type requires special
+handling if there are changes to its encrypted attributes or attributes that are included in AAD - in both cases an object must be re-encrypted. For this case the
+ESO Service exposes a Model Version wrapper function API `createModelVersion`. Similar in utility to its predecessor (`createMigration` - used with non-Model Version
+Saved Object legacy migrations), `createModelVersion` provides a way to wrap a Model Version definition such that decryption and encryption occur automatically
+during migration of any applicable ESOs.
+
+In addition to a Model Version definition, `createModelVersion` also requires both an "input type" and "output type" `EncryptedSavedObjectTypeRegistration`
+input parameters.
+
+```ts
+export interface CreateEsoModelVersionFnOpts {
+  modelVersion: SavedObjectsModelVersion;
+  shouldTransformIfDecryptionFails?: boolean;
+  inputType: EncryptedSavedObjectTypeRegistration;
+  outputType: EncryptedSavedObjectTypeRegistration;
+}
+```
+
+The `inputType` parameter provides the necessary ESO registration definition for decrypting an ESO of the preceding Model Version prior to performing any transforms
+defined by the Model Version. The `outputType` parameter provides the necessary ESO registration definition for encrypting an ESO once the transforms defined in the
+Model Version have been completed. All of the Model Version transform functions ('unsafe_transform', 'data_backfill', 'data_removal'), are merged into a single
+transform function for efficiency. This way, each ESO only needs to be decrypted and re-encrypted once to incorporate all of the changes defined in a Model Version.
+The optional `shouldTransformIfDecryptionFails` parameter defines whether an ESO type should proceed with the Model Version changes even if decryption fails.
+
+Some examples of `createModelVersion` can be found in the ESO Model Version example plugin (examples/eso_model_version_example/server/plugin.ts) [ToDo: better link?]
+
+For more information regarding Model Versions see the documentation at packages/core/saved-objects/core-saved-objects-server/docs/model_versions.md [ToDo: not sure
```

Comment:
Looks like we have some docs for model version here: https://docs.elastic.dev/kibana-dev-docs/tutorials/saved-objects#defining-model-versions